src/components/StoreMigrator/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { addressBookMigrate } from 'src/logic/addressBook/store/actions'
+import { logError, Errors } from 'src/logic/exceptions/CodedException'
+import { MigrationMessage } from 'src/routes/migration/container'
+import { MIGRATION_ADDRESS } from 'src/routes/routes'
+
+const MAINET_URL = 'https://pr2695--safereact.review.gnosisdev.com/mainnet/app'
+const networks = [
+  {
+    safeUrl: 'https://pr2695--safereact.review.gnosisdev.com/polygon/app',
+  },
+  {
+    safeUrl: 'https://pr2695--safereact.review.gnosisdev.com/rinkeby/app',
+  },
+]
+
+type MigrationMessageEvent = MessageEvent & {
+  data: MigrationMessage
+}
+
+const StoreMigrator: React.FC = () => {
+  const [currentNetwork, setCurrentNetwork] = useState(0)
+  const dispatch = useDispatch()
+  //let networks = getNetworks()
+
+  // Recieve the data to be migrated and save it into the localstorage
+  useEffect(() => {
+    const saveEventData = (event: MigrationMessageEvent) => {
```

Comment:
Please test this exact function, it has more logic than just the address book import.